datahub/dataset/views.py

```diff
@@ -70,3 +84,121 @@ def get_dataset(self):
         )
+
+
+class InvestmentProjectsDatasetView(HawkResponseSigningMixin, APIView):
+    """
+    An APIView that provides 'get' action which queries and returns desired fields for
+    Investment Projects Dataset to be consumed by Data-flow periodically.
+    Data-flow uses response result to insert data into Dataworkspace through its defined
+    API endpoints. The goal is presenting various reports to the users out of flattened table
+    and let analyst to work on denormalized table to get more meaningful insight.
+    """
+
+    authentication_classes = (HawkAuthentication, )
+    permission_classes = (HawkScopePermission, )
+    required_hawk_scope = HawkScope.data_flow_api
+    pagination_class = InvestmentProjectsDatasetViewCursorPagination
+
+    def get(self, request):
+        """Endpoint which serves all records for Investment Projects Dataset"""
+        dataset = self.get_dataset()
+        paginator = self.pagination_class()
+        page = paginator.paginate_queryset(dataset, request, view=self)
+        return paginator.get_paginated_response(page)
+
+    def get_dataset(self):
+        """Returns list of Investment Projects Dataset records"""
+        return InvestmentProject.objects.annotate(
+            actual_uk_region_names=get_investment_project_to_many_string_agg_subquery(
+                'actual_uk_regions__name',
+            ),
+            business_activity_names=get_investment_project_to_many_string_agg_subquery(
+                'business_activities__name',
+            ),
+            competing_countries=get_aggregate_subquery(
+                InvestmentProject,
+                ArrayAgg('competitor_countries__id', ordering=('competitor_countries__id',)),
+            ),
+            delivery_partner_names=get_investment_project_to_many_string_agg_subquery(
+                'delivery_partners__name',
+            ),
+            investor_company_sector=get_sector_name_subquery('investor_company__sector'),
+            level_of_involvement_name=get_empty_string_if_null_expression(
+                'level_of_involvement__name',
+            ),
+            project_reference=get_project_code_expression(),
+            strategic_driver_names=get_investment_project_to_many_string_agg_subquery(
+                'strategic_drivers__name',
+            ),
+            sector_name=get_sector_name_subquery('sector'),
+            team_member_ids=get_aggregate_subquery(
+                InvestmentProject,
+                ArrayAgg('team_members__adviser_id', ordering=('team_members__id',)),
+            ),
+            uk_company_sector=get_sector_name_subquery('uk_company__sector'),
+        ).values(
+            'actual_land_date',
+            'actual_uk_region_names',
+            'allow_blank_possible_uk_regions',
+            'anonymous_description',
+            'archived',
+            'archived_by_id',
+            'archived_on',
+            'associated_non_fdi_r_and_d_project__id',
```

Comment:
Should we make this:

```suggestion
            'associated_non_fdi_r_and_d_project_id',
```